### PR TITLE
Leave room btn

### LINKED ISSUE #62: Leave room functionality

### PR DESCRIPTION
**Pull request by GreyMatteOr**
## Description

Display a current count of users that are online currently.
Create an input for a user-name (default of guest) and also display your username at the top.
Display all usernames of all players that are in your game room.
Added a `leave room` button that will be displayed after you join a room.
Correct display based on the various states that might be returned from the server when leaving a game or disconnecting.

## Context

closes #62 

## Where To Start

Mostly in App, then trace through lobby and gamepage

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] live with multiple clients on a local server

# Notes

Still need to style.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -125,6 +125,7 @@ export class App extends Component{
                 attempts={this.state.attempts}
                 display={this.state.display}
                 isGenerator={this.state.isGenerator}
+                playerNames={this.state.playerNames}
```
Comment (holladayian) [inline src/App.js:144]: Nice! We were just talking about this! Looks good

File: src/Gamepage/Gamepage.js [hunk 2]
```diff
@@ -67,20 +66,38 @@ export class Gamepage extends Component {
...
+          <h2>
+            <em>Chat</em>
+          </h2>
         </div>
         <div className="board">
           <div className="draw-board">
-            <h2><em>Draw Board</em></h2>
-          <div className="display-word">{this.splitDisplay()}</div>
+            <h2>
+              <em>Draw Board</em>
```
Comment (holladayian) [inline src/Gamepage/Gamepage.js]: We should probably remove `Draw Board` if we're not going to implement drawing by the due date

File: src/Homepage/Homepage.css [hunk 3]
```diff
@@ -23,7 +23,12 @@
...
 
 .generator:hover{
-  background-color: #BF8620
+  background-color: #BF8620;
+}
+
+.generator:disabled {
+  background-color: #7b4200;
+  color: #404040;
 }
```
Comment (holladayian) [inline src/Homepage/Homepage.css:32]: Strange. I thought we already had this functionality.

File: src/Lobby/Lobby.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import './Lobby.css';
+// import './Lobby.css';
```
Comment (holladayian) [inline src/Lobby/Lobby.js:3]: Let's remember to comment this back in so lobby can have styles!

**Review by holladayian — APPROVED** [review]

File: src/Gamepage/Gamepage.js (hunk 2 above)
Comment (GreyMatteOr) [reply src/Gamepage/Gamepage.js]: True! It might be cumbersome if each player will have their own display, too...

**Pull request merged**